crates/holochain_cascade/tests/tests/get_record.rs

```diff
@@ -0,0 +1,307 @@
+use ::fixt::fixt;
+use holo_hash::HashableContentExtSync;
+use holochain_cascade::CascadeImpl;
+use holochain_state::prelude::{
+    insert_op_authored, insert_op_cache, insert_op_dht, set_validation_status, set_when_integrated,
+    test_authored_db, test_cache_db, test_dht_db,
+};
+use holochain_types::{
+    dht_op::{ChainOp, DhtOp},
+    fixt::{
+        CreateFixturator, CreateLinkFixturator, DeleteFixturator, DeleteLinkFixturator,
+        SignatureFixturator, UpdateFixturator,
+    },
+    prelude::{
+        AppEntryDef, EntryType, EntryVisibility, GetOptions, RecordEntry, Timestamp,
+        ValidationStatus,
+    },
+};
+use holochain_zome_types::{Action, Entry};
+
+#[tokio::test(flavor = "multi_thread")]
+async fn get_action_from_authored() {
+    let authored = test_authored_db();
+    let cascade = CascadeImpl::empty().with_authored(authored.clone().into());
+
+    let create_action = Action::Create(fixt!(Create));
+    let create_op = DhtOp::ChainOp(Box::new(ChainOp::StoreRecord(
+        fixt!(Signature),
+        create_action.clone(),
+        RecordEntry::NA,
+    )))
+    .into_hashed();
+    let op_hash = create_op.hash.clone();
+
+    authored.test_write(move |txn| {
+        insert_op_authored(txn, &create_op).unwrap();
+    });
+    // Get should not return the op while it's not valid and integrated.
+    let maybe_record = cascade
+        .dht_get(create_action.to_hash().into(), GetOptions::local())
+        .await
+        .unwrap();
+    assert!(maybe_record.is_none());
+
+    // Set op to valid.
+    let op_hash2 = op_hash.clone();
+    authored.test_write(move |txn| {
+        set_validation_status(txn, &op_hash2, ValidationStatus::Valid).unwrap();
+    });
+    // Get should not return the op while it's not integrated.
+    let maybe_record = cascade
+        .dht_get(create_action.to_hash().into(), GetOptions::local())
+        .await
+        .unwrap();
+    assert!(maybe_record.is_none());
+
+    // Set op to integrated.
```

Comment:
Maybe a code comment here to justify this and the flag change above? It's still weird to me that we have to set these flags in the authored database and so it'd help when reading this test to have a hint that this is what the code is required to do for queries against the authored database to work?